Expensify/App
# [No QA] Merge TextInputWithName to our TextInput component

### Details

- [x] **TextInputWithName**
    #### Purpose of component
    1. Support name attribute on Web brower. 
    
    The same behavior can be easily moved to `TextInput`. 
    
### Fixed Issues

$ https://github.com/Expensify/App/issues/6584

### Tests
1. Open the Login page on the web.
2. Open Inspect Element from DevTools. 
3. Select Input element. 
4. Check that the Name attribute is correctly set as provided via prop.



### Tested On


### Screenshots


#### Web | Desktop | Mobile Web

![image](https://user-images.githubusercontent.com/24370807/153769662-057e3157-2405-4a84-8c3a-fb900a181e4b.png)

#### iOS

![image](https://user-images.githubusercontent.com/24370807/153769715-fb4b87d7-add3-41f3-a06d-814b281e80d1.png)

#### Android

![image](https://user-images.githubusercontent.com/24370807/153883428-2938914c-7111-4961-b86e-086c72b87974.png)
